src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -30,6 +30,16 @@ internal static bool IsDirectorySeparator(char c)
             return c == DirectorySeparatorChar || c == AltDirectorySeparatorChar;
         }
 
+        internal static string TrimTrailingSeparators(string s)
+        {
+            while (s.Length > 0 && IsDirectorySeparator(s[s.Length - 1]))
+            {
+                s = s.Substring(0, s.Length - 1);
```

Comment:
Consider calculating where to truncate and truncating once rather than in a loop.